cmd/check-markdown-rs/src/main.rs

```diff
@@ -0,0 +1,317 @@
+use comrak::arena_tree::Node;
+use comrak::{
+    nodes::{Ast, NodeValue},
+    parse_document, Arena, ComrakOptions,
+};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::env;
+use std::fs;
+use url::Url;
+
+// Main function
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let arena = Arena::new();
+    let args: Vec<String> = env::args().collect();
+
+    if args.len() < 2 {
+        eprintln!("Usage: {} <input_file>", args[0]);
+    }
```

Comment:
- You need to call ` std::process::exit(1)` after displaying the message: currently, the message is displayed but the program keeps running, then fails:
   ```bash
   $ check-markdown-rs
   Usage: check-markdown-rs <input_file>
   thread 'main' panicked at 'index out of bounds: the len is 1 but the index is 1', src/main.rs:22:52
   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
   ```
- Please can you add some argument parsing code so that the user can run:
  ```bash
  $ check-markdown-rs --help
  $ check-markdown-rs --version
  ```